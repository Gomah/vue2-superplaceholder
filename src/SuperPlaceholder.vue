<template>
  <input
    :value="value"
    placeholder=""
    @input="updateValue($event.target.value)"
  />
</template>
<script>
export default {
  props: {
    value: {
      type: String,
      default: null,
    },
    sentences: {
      type: Array,
      default: () => [],
    },
    options: {
      type: Object,
      default: () => ({}),
    },
    selector: {
      default: null,
    },
  },

  data() {
    return {
      superplaceholder: null,
    };
  },

  methods: {
    updateValue(value) {
      this.$emit('input', value);
    },
  },

  mounted() {
    const superplaceholder = require('superplaceholder');

    this.superplaceholder = superplaceholder({
      el: this.selector || this.$el,
      options: this.options,
      sentences: this.sentences,
    });
  },

  beforeDestroy() {
    this.superplaceholder.destroy();
  },
};
</script>
