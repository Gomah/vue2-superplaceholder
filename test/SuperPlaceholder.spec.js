import { shallowMount } from '@vue/test-utils';
import SuperPlaceholder from '../src/SuperPlaceholder.vue';

describe('SuperPlaceholder', () => {
  test('mounts properly', () => {
    const wrapper = shallowMount(SuperPlaceholder);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = shallowMount(SuperPlaceholder);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
