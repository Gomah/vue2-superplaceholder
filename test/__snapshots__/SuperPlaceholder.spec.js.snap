// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuperPlaceholder renders properly 1`] = `<input placeholder="">`;
